### PR TITLE
refactor: explanations for interfaces

### LINKED ISSUE #39: contracts/interfaces cleanup
add more description also

### PR DESCRIPTION
**Pull request by bogdan-manole**
Closes #39

### CODE REVIEW

**Comment by bogdan-manole** [comment]
Should we change the filename  for contracts/interface/README.md ?

**Comment by kenodressel** [comment]
Why, whats wrong with the filename?

**Comment by bogdan-manole** [comment]
> Why, whats wrong with the filename?

I'm just wondering: is it wrong having multiple README.md on different folder levels ?

File: contracts/interfaces/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+- The interfaces from ./for-export are autogenerated and are full contracts definition interface to be
```
Comment (ghallak) [inline contracts/interfaces/README.md:1]:
```suggestion
- The interfaces from ./for-export are auto-generated and are full contracts definition interface to be
```

File: contracts/interfaces/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+used by sdk-clients outside this project in order to interact with the dex.
+In js project these can be accessed in the form of `var source = require('dex-contracts-v2/build/INTERFACE_NAME.aes')`
```
Comment (ghallak) [inline contracts/interfaces/README.md:3]:
```suggestion
In a JavaScript project these can be accessed in the form of `var source = require('dex-contracts-v2/build/INTERFACE_NAME.aes')`
```

File: contracts/interfaces/README.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+
+- The the purpose for the rest of the interfaces is to be used inside this project for tests (js) or inside the contracts.
```
Comment (ghallak) [inline contracts/interfaces/README.md:6]:
```suggestion
- The purpose of the rest of the interfaces is to use them inside this project for the tests (JS) or inside the contracts.
```

File: contracts/interfaces/README.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+We are defining here only the entrypoints that are used and sometime we splitt a contract entrypoints into different interface files
```
Comment (ghallak) [inline contracts/interfaces/README.md:7]:
```suggestion
We are defining here only the entrypoints that are used and sometime we split a contract entrypoints into different interface files 
```

File: contracts/interfaces/README.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+- The interfaces from ./for-export are autogenerated and are full contracts definition interface to be
+used by sdk-clients outside this project in order to interact with the dex.
+In js project these can be accessed in the form of `var source = require('dex-contracts-v2/build/INTERFACE_NAME.aes')`
+
+
+- The the purpose for the rest of the interfaces is to be used inside this project for tests (js) or inside the contracts.
+We are defining here only the entrypoints that are used and sometime we splitt a contract entrypoints into different interface files 
+to avoid circular references (eg. IAedexV2Factory and IAedexV2FactoryForPair)
+More details you can find it at the top level of these interfaces
```
Comment (ghallak) [inline contracts/interfaces/README.md:9]:
```suggestion
You can find further details at the top of each of these interfaces
```

File: contracts/interfaces/IAEX9Minimal.aes [hunk 6]
```diff
@@ -1,5 +1,7 @@
 @compiler >= 6
 
+//Interface for tokens and pair with minimal entrypoints.
```
Comment (ghallak) [inline contracts/interfaces/IAEX9Minimal.aes:3]:
```suggestion
// Interface for tokens and pair with minimal entrypoints.
```

Comment (ghallak) [inline contracts/interfaces/IAEX9Minimal.aes:3]: I don't get what you mean by `pair with minimal entrypoints`.

**Review by ghallak — CHANGES_REQUESTED** [review]

**Review by thepiwo — APPROVED** [review]

**Pull request merged**